spring-ai-core/src/main/java/org/springframework/ai/chat/history/ChatMemory.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2024-2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.ai.chat.history;
+
+import java.util.List;
+
+import org.springframework.ai.chat.messages.Message;
+
+/**
+ * @author Christian Tzolov
+ */
+public interface ChatMemory {
+
+	default void add(String conversationId, Message messages) {
+		this.add(conversationId, List.of(messages));
+	}
+
+	void add(String conversationId, List<Message> messages);
+
+	List<Message> get(String conversationId);
```

Comment:
```suggestion
	List<Message> get(String conversationId, int lastN);
```
the limit will want to be a pushed-down predicate, e.g. in non-memory impls.

Reply:
Agree. Current ChatMemory is likely very naive and the moment we try to implemnt a Repository around it I guess more changes will come.